workflow-to-js-transformer/src/main/resources/Workflow.xslt

```diff
@@ -266,25 +272,61 @@
+                return ALREADY_EXECUTED;
+            }
+            // propagate the post-processing field to response custom data if it exists on the document custom data
+            var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
+            if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
+                responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            }
 
-        // propagate the post-processing field to response custom data if it exists on the document custom data
-        var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
-        if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
-        responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            // Update document destination queue to that specified by action and pass appropriate settings and customData
+            var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
+            var responseOptions = document.getTask().getResponseOptions();
+            responseOptions.setQueueName(queueToSet);
+            responseOptions.setCustomData(responseCustomData);
+            return ACTION_TO_EXECUTE;
         }
 
-        // Update document destination queue to that specified by action and pass appropriate settings and customData
-        var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
-        var responseOptions = document.getTask().getResponseOptions();
-        responseOptions.setQueueName(queueToSet);
-        responseOptions.setCustomData(responseCustomData);
+        // executes the field mapping action on the document
+        function executeFieldMapping(document, mappings){
+        // get the field values to map (from the document)
+        var documentFieldsValuesMap = {};
+        for(var mappingKey in mappings) {
+        var documentFieldToMapFrom =  document.getField(mappingKey);
+        if(!documentFieldToMapFrom.hasValues()){
+        continue;
+        }
+        documentFieldsValuesMap[mappingKey] = documentFieldToMapFrom.getValues();
+        documentFieldToMapFrom.clear();
         }
 
-    </xsl:template>
+        // for each mapping add the original field value with the new key
+        for(var mappingKey in mappings) {
+        var mappingDestination = mappings[mappingKey];
+        var documentFieldToMapTo = document.getField(mappingDestination);
+        for each(var fieldValue in documentFieldsValuesMap[mappingKey]) {
+        if(fieldValue.isReference()){
+        documentFieldToMapTo.addReference(fieldValue.getReference());
+        continue;
+        }
+        documentFieldToMapTo.add(fieldValue.getValue());
+        }
+        }
+        }
```

Comment:
Why is there no indenting to make this readable?

Reply:
Indenting has been added before and then lost. It looks to be the fault of the IDE which sees the file as XSLT and then treats the additional indention of text (which it doesn't know is JavaScript) inside that as unnecessary. Can see this is if you go to the file, indent and the run formatting (ALT-SHIFT-F) in Netbeans. InteliJ does this also, looking into away to avoid it.

Reply:
I think as long as people don't run autoformat on the file it will maintain the spacing. I've indented the JS to make it more readable and committed it.